sktime/classifiers/proximity.py

```diff
@@ -0,0 +1,1084 @@
+    Attributes
+    ----
+    exemplar_instances : panda dataframe
+        the chosen exemplar instances
+    exemplar_class_labels : numpy 1d array
+        array of class labels corresponding to the exemplar instances
+        the exemplar instances class labels
+    remaining_instances : panda dataframe
+        the remaining instances after exemplars have been removed
+    remaining_class_labels : numpy 1d array
+        array of class labels corresponding to the exemplar instances
+        the remaining instances class labels after picking exemplars
+    branch_instances : list of panda dataframes
+        list of dataframes of instances, one for each child of this stump. I.e. if a stump splits into two children,
+        there will be a list of dataframes of length two. branch_instance[0] will contain all train instances
+        closest to exemplar 0, branch_instances[1] contains all train instances closest to exemplar 1,
+        etc. Exemplars are in the exemplar_instances variable
+    branch_class_labels: list of numpy 1d arrays
+        similar to branch_instances, but contains the class labels of the instances closest to each exemplar
+    distance_measure_param_perm: dict
+        parameters to pass to the distance measure method
+    distance_measure: callable
+        the distance measure to use for measure similarity between instances
+    gain: float
+        the gain of this stump
+    label_encoder : LabelEncoder
+        a label encoder, can be pre-populated
+    classes_ :
+        pointer to the label_encoder classes_
+    '''
+
+    __author__ = 'George Oastler (linkedin.com/goastler)'
+
+    def __init__(self,
+                 pick_exemplars_method = None,
+                 param_perm = None,
+                 gain_method = None,
+                 label_encoder = None,
+                 rand = np.random.RandomState):
+        super().__init__(rand = rand)
+        self.param_perm = param_perm
+        self.gain_method = gain_method
+        self.pick_exemplars_method = pick_exemplars_method
+        # vars set in the fit method
+        self.exemplar_instances = None
+        self.exemplar_class_labels = None
+        self.remaining_instances = None
+        self.remaining_class_labels = None
+        self.branch_instances = None
+        self.branch_class_labels = None
+        self.distance_measure_param_perm = None
+        self.distance_measure = None
+        self.gain = None
+        self.label_encoder = label_encoder
+        self.classes_ = None
+
+    @staticmethod
+    def get_distance_measure_key():
+        '''
+        get the key for the distance measure. This key is required for picking the distance measure out of the
+        param_perm constructor parameter.
+        ----
+        Returns
+        ----
+        key : string
+            key for the distance measure for the param_perm dict
+        '''
+        return 'dm'
+
+    def fit(self, instances, class_labels, should_check_data = True):
```

Comment:
To make it consistent, rename: 
* instances to X
* class_labels to y
* should_check_data to input_checks